SlackClone/SlackChatMessageContentView.swift

```diff
@@ -7,242 +7,27 @@ import StreamChatUI
-                textView!.topAnchor.constraint(equalTo: messageBubbleView!.topAnchor),
-                
-                attachmentsView!.topAnchor.constraint(equalTo: textView!.bottomAnchor, constant: 5),
-                attachmentsView!.leadingAnchor.constraint(equalTo: messageBubbleView!.leadingAnchor),
-                attachmentsView!.trailingAnchor.constraint(equalTo: messageBubbleView!.trailingAnchor),
-                attachmentsView!.bottomAnchor.constraint(equalTo: messageBubbleView!.bottomAnchor, constant: -5)
-            ])
-        case [.text]:
-            NSLayoutConstraint.activate([
-                textView!.leadingAnchor.constraint(equalTo: messageBubbleView!.leadingAnchor),
-                textView!.trailingAnchor.constraint(equalTo: messageBubbleView!.trailingAnchor),
-                textView!.topAnchor.constraint(equalTo: messageBubbleView!.topAnchor),
-                textView!.bottomAnchor.constraint(equalTo: messageBubbleView!.bottomAnchor)
-            ])
-        default:
-            NSLayoutConstraint.activate([
-                textView!.leadingAnchor.constraint(equalTo: messageBubbleView!.leadingAnchor),
-                textView!.trailingAnchor.constraint(equalTo: messageBubbleView!.trailingAnchor),
-                textView!.topAnchor.constraint(equalTo: messageBubbleView!.topAnchor),
-                textView!.bottomAnchor.constraint(equalTo: messageBubbleView!.bottomAnchor)
-            ])
-        }
+    override func createTextView() -> UITextView {
+        let textView = super.createTextView()
+        textView.textContainerInset = .zero
+        return textView
     }
 }
 
-extension _ChatMessageGroupPart {
-    var textContent: String {
-        guard message.type != .ephemeral else {
-            return ""
-        }
-        
-        guard message.deletedAt == nil else {
-            return "Message was deleted"
-        }
-        
-        return message.text
-    }
-    
-    var layoutOptions: ChatMessageContentViewLayoutOptions {
-        guard message.deletedAt == nil else {
-            return [.text]
-        }
-        
-        var options: ChatMessageContentViewLayoutOptions = []
-        
-        if !textContent.isEmpty {
-            options.insert(.text)
-        }
-        
-        if quotedMessage != nil {
-            options.insert(.quotedMessage)
-        }
-        
-        if message.attachments.contains(where: { $0.type == .image || $0.type == .giphy || $0.type == .file }) {
-            options.insert(.attachments)
-        } else if message.attachments.contains(where: { $0.type.isLink }) {
-            // link preview is visible only when no other attachments available
-            options.insert(.linkPreview)
-        }
-        
-        return options
+extension ContainerStackView {
+    func changeOrdering() {
+        let subviews = self.subviews
+        removeAllArrangedSubviews()
+        addArrangedSubviews(subviews.reversed())
```

Comment:
That feels handy, how about making it part of `ContainerStackView` public API? 😄

Reply:
@evsaev I'm OK with moving to SDK but I'd keep it internal for now :)